Short-Answer/Algorithms_Answers.md

```diff
@@ -1,15 +1,19 @@
-#### Please add your answers to the ***Analysis of  Algorithms*** exercises here.
+#### Please add your answers to the **_Analysis of Algorithms_** exercises here.
 
 ## Exercise I
 
-a)
+a) The runtime of this is O(n) because the number of operations increases directly in proportion to the size of n. Although the loop involves an upper limit of n^3, it increments by n^2, and n^3/n^2 = n. For example, with n = 2, teh while loop runs twice. If we double n to n = 4, the number of times the loop runs doubles to 4, also. If we again double n to n = 8, the loop runs 8 times. This shows O(n) runtime.
 
+b) The runtime of this is O(n^2) because the big O notation for the outer loop is O(n), and the big O notation for the inner loop is O(n). When loops are nested, we multiply their big O notations together, which gives us O(n) \* O(n) = O(n^2)
```

Comment:
For this one we have to consider both loops individually:
The outer for loop will run n times, since it is incrementing i by 1 each iteration. This is O(n).
The inner loop is doubling the value of j every time it runs. This means that the number of times the inner while loop runs is not growing at a linear rate. This is O(log(n)).
Since these loops are nested, we multiply the individual runtimes of each loop to get the runtime of the entire block.
So the answer is O(n log(n))